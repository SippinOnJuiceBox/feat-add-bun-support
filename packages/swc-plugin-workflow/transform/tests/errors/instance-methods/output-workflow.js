export class TestClass {
  // Error: instance methods can't have directives
  async instanceMethod() {
    'use step';
    return 'not allowed';
  }
  // Error: instance methods can't have workflow directive either
  async anotherInstance() {
    'use workflow';
    return 'also not allowed';
  }
  // This is ok - static methods can have directives
  static async staticMethod() {
    'use step';
    return 'allowed';
  }
}
