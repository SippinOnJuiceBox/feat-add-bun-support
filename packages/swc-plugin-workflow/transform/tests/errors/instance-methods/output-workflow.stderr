  x Instance methods cannot be marked with "use step". Only static methods, functions, and object methods are supported.
   ,-[input.js:3:1]
 2 |       // Error: instance methods can't have directives
 3 | ,->   async instanceMethod() {
 4 | |       'use step';
 5 | |       return 'not allowed';
 6 | `->   }
   `----
  x Instance methods cannot be marked with "use workflow". Only static methods, functions, and object methods are supported.
    ,-[input.js:9:1]
  8 |       // Error: instance methods can't have workflow directive either
  9 | ,->   async anotherInstance() {
 10 | |       'use workflow';
 11 | |       return 'also not allowed';
 12 | `->   }
    `----
